src/DynamoPackagesWPF/PackageManagerViewExtension.cs

```diff
@@ -64,9 +81,15 @@ public void Dispose()
         public void Loaded(ViewLoadedParams viewLoadedParams)
         {
             RequestLoadLayoutSpecs(packageManager.PackageLoader.LocalPackages);
+            var packagesToCheck = packageManager?.PackageLoader.LocalPackages;
+            if(packagesToCheck != null)
+            {
+                RaisePackageHostNotifications(packagesToCheck);
+            }
```

Comment:
Can you explain again why are we raising them here again?

Reply:
This hook (`IViewExtension.Loaded`) occurs when the `PackageManagerViewExtension` is loaded, this is basically at startup time, so there are a set of packages that were just loaded that we need to check. 

`PackageLoaded_Handler` will handle late loaded extensions. Since this is a view extension, this class is not even loaded until *after* the packageManager finishes loading the packages in the users's package locations at startup.

It's the same pattern for the other things this extension checks, layoutspecs, view extensions etc.

Reply:
Won't the `packagesToCheck` already be checked for hosts (using `CheckIfPackagesTargetOtherHosts`) by the package manager when it's loading the packages for the first time? If not, shouldn't we be doing the check at package load time?

Reply:
No it won't be done by the package manager at package load time, the method is in that class so someone can use it form there, but I decided to defer it in this case, because at that point the view is not loaded so notifications / message boxes will not be able to be raised.

I put these checks here so that we ensure the view is loaded, that means other extensions like `NotificationsViewExtension` are ready to actually log notifications.

If we move some of the checks to `packageManagerExtension` then we will need to build up a list of notifications and send them to NotificationsViewExtension at some point later.

I can do that, but it seemed simpler to have all the checks done here - unless you think it is important to do these checks in headless mode as well - the big caveat being, the `NotificationViewExtension` is not loaded in headless mode so sending notifications won't do much! It will still log.

Reply:
Understood, thanks for the explanation. I overlooked the fact that none of the view extensions would be ready at that point in time. I'm okay with this approach.